lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -885,6 +929,29 @@ def create_autoscaling_group(connection):
             ag['LoadBalancerNames'] = load_balancers
         if target_group_arns:
             ag['TargetGroupARNs'] = target_group_arns
+        if launch_config_name:
+            try:
+                launch_configs = describe_launch_configurations(connection, launch_config_name)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json(msg="Failed to describe launch configurations",
+                                 exception=traceback.format_exc())
+            if len(launch_configs['LaunchConfigurations']) == 0:
+                module.fail_json(msg="No launch config found with name %s" % launch_config_name)
+            ag['LaunchConfigurationName'] = launch_configs['LaunchConfigurations'][0]['LaunchConfigurationName']
+        elif launch_template:
+            try:
+                lt = describe_launch_templates(ec2_connection, launch_template)['LaunchTemplates'][0]
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json(msg="Failed to describe launch templates",
+                                 exception=traceback.format_exc())
+            if not lt:
```

Comment:
Can this happen? Rather than how the describe_launch_configuration call returns an empty list if the launch config doesn't exist, the boto3 describe_launch_templates call appears to raise an exception like `InvalidLaunchTemplateName.NotFoundException` when the template doesn't exist.